xayn-ai/src/embedding/utils.rs

```diff
@@ -57,23 +57,32 @@ where
 ///
 /// # Panics
 /// Panics if the vectors don't consist solely of real values or their shapes don't match.
+///
+/// # Safety
+/// Only use this with [`TrustedLen`] iterators. The bound can't be named currently, because the
+/// trait is nightly-gated. [`ExactSizeIterator`] isn't a feasible replacement, because it isn't
```

Comment:
> [`ExactSizeIterator`] isn't a feasible replacement,

It's also not "trusted", i.e. it must be safe for it to lie about it's length.

Reply:
updated the docs